app/models/exercise/representation.rb

```diff
@@ -22,7 +24,8 @@ class Exercise::Representation < ApplicationRecord
   # This is too inefficient. Get the representations and then their submissions instead.
   # has_many :submission_representation_submissions, through: :submission_representations, source: :submission
 
-  has_many :published_solutions, foreign_key: "published_exercise_representation", class_name: "Solution",
+  has_many :published_solutions, -> { where(status: :published) },
+    foreign_key: "published_exercise_representation", class_name: "Solution",
```

Comment:
This was a bug that meant we counted all solutions, not just published ones in this field. So we'll need to recalculate and resync all representations to the indexes anyway.